rhino/src/main/java/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -1742,6 +1736,43 @@ public static final class DescriptorInfo {
             value = getProperty(desc, "value");
             accessorDescriptor = getter != NOT_FOUND || setter != NOT_FOUND;
         }
+
+        DescriptorInfo(
+                Object enumerable,
+                Object writable,
+                Object configurable,
+                Object getter,
+                Object setter,
+                Object value) {
+            this.enumerable = enumerable;
```

Comment:
I know it makes the code bigger but would you consider a "builder" pattern for this? I worry about being able to read the code when every DescriptorInfo constructor has six parameters of the same type in a row.